master/lustregittagpoller.py

```diff
@@ -0,0 +1,122 @@
+# -*- python -*-
+# ex: set syntax=python:
+
+import itertools
+import os
+
+from buildbot import config
+from twisted.internet import defer
+from twisted.internet import utils
+from twisted.python import log
+from buildbot.util import epoch2datetime
+from buildbot.changes.gitpoller import GitPoller 
+
+class LustreTagPoller(GitPoller):
+
+    """This source polls a git repo for new tags and generates a change.
+
+    This source will poll a remote git repository for new tags and finds changes
+    to existing tags. Once a new or changed tag is found, a change is submitted to
+    the build master. The poller maintains a set of previously seen tags and each
+    tags associated revision (SHA). On each poll iteration, all references are
+    checked against the prior set. A tag is considered new if it doesn't exist in
+    the set. A tag is considered change when the tag has been seen but its revision
+    has changed. The set is maintain in the buildbot master's database."""
+
+    def __init__(self, **kwargs):
+        if kwargs.get('workdir') is None:
+            kwargs['workdir'] = 'lustretagpoller-work'
+
```

Comment:
Some of the arguments in the parent class no longer make sense, and should be caught here, and an exception raised.  "branch" and "branches" are invalid options to this subclass, right?  Go through the rest of the options and see if any others are now invalid.  Note exceptions in the comment above.